dpnp/dpnp_iface_statistics.py

```diff
@@ -451,16 +455,55 @@ def _get_padding(a_size, v_size, mode):
     return l_pad, r_pad
 
 
-def _run_native_sliding_dot_product1d(a, v, l_pad, r_pad):
+def _choose_conv_method(a, v, rdtype):
+    assert a.size >= v.size
+    if rdtype == dpnp.bool:
+        return "direct"
+
+    if v.size < 10**4 or a.size < 10**4:
+        return "direct"
+
+    if dpnp.issubdtype(rdtype, dpnp.integer):
+        max_a = int(dpnp.max(dpnp.abs(a)))
+        sum_v = int(dpnp.sum(dpnp.abs(v)))
+        max_value = int(max_a * sum_v)
+
+        default_float = dpnp.default_float_type(a.sycl_device)
+        if max_value > 2 ** numpy.finfo(default_float).nmant - 1:
+            return "direct"
```

Comment:
It seems there is not any test covering that line in public CI. Is there any way to add it?

Reply:
It is hard to add test for it.
Probably it is better to have separate tests for `_choose_conv_method` as scipy do. But it has [choose_conv_method](https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.choose_conv_method.html#scipy.signal.choose_conv_method) as part of public API which actually used only inside `correlate`/`convolve` functions

Reply:
I believe the current set of tests is more than enough then.